stdlib/effect.mli

```diff
@@ -16,10 +16,7 @@
 
     @since 5.0 *)
 
-type _ t = ..
-(** The type of effects. *)
```

Comment:
I wonder if we could still keep a type alias?
```ocaml
type 'a t = 'a eff = ..
```
I don't have a strong opinion, but this seems to be compatible with other instances in the standard library (e.g., `Bool.t = bool`) thought technically speaking it's `Effect.t` not `Eff.t`. More importantly, existing libraries using `Effect.t` can compile without changes.

Reply:
I thought about this a bit, but since OCaml 5 is not released yet, any library that currently uses OCaml 5 should expect breaking changes. I am not fond of compatibility fixes even before we have made a release where we've committed to not breaking things. :-) 

There are only a few libraries in the wild using effects today -- eio, domainslib and a couple of tutorials. The change is mechanical in nature and I'm sure there is a clever enough sed script that fixes 90% of the cases.

Reply:
Well, we also sent a few serious libraries to OPAM recently that use effects, and there are others :wink: but yes, we can quickly adapt and release fixed versions to OPAM.

Reply:
> Well, we also published a few serious libraries 

Cool! Can you point me to them? I'm interested to learn about the uses of effects. 

I still think we should not add an `Effect.t` type. We've made no commitment yet to not break the API and the compiler maintainers should exercise that to avoid introducing and carrying around compatibility code.

Reply:
I am fine with either way. :-) Speaking of our recent libraries...

- [algaeff: Reusable Effects-Based Components](https://github.com/RedPRL/algaeff) (I don't want to claim there's any novelty in this one---I think we were just packaging well-know patterns into a convenient OPAM package.)
- [Yuujinchou 友人帳: Name Modifiers](https://github.com/RedPRL/yuujinchou), see my [TYPES 2022 talk](https://www.youtube.com/watch?v=5Cp8jNrWSjI)

We are also building a fun proof assistant [algaett](https://github.com/RedPRL/algaett) using these libraries, but we probably will not send it to OPAM, ever :laughing:

Reply:
I would  be in favor of re-exporting the effect type in the `Effect` module: this is what is done for all other predefined types.

Reply:
Done.